File: mambo-shared/mambo-protocol/mambo-protocol-client/src/main/java/org/mambo/protocol/client/messages/PartyCancelInvitationMessage.java


// Generated on 01/04/2013 14:54:29
package org.mambo.protocol.client.messages;

import java.util.*;
import org.mambo.protocol.client.types.*;
import org.mambo.protocol.client.enums.*;
import org.mambo.protocol.client.*;
import org.mambo.core.io.*;

public class PartyCancelInvitationMessage extends AbstractPartyMessage {
    public static final int MESSAGE_ID = 6254;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int guestId;
    
    public PartyCancelInvitationMessage() { }
    
    public PartyCancelInvitationMessage(int partyId, int guestId) {
        super(partyId);
        this.guestId = guestId;
    }
    
    @Override
    public void serialize(DataWriterInterface writer) {
        super.serialize(writer);
        writer.writeInt(guestId);
    }
    
    @Override
    public void deserialize(DataReaderInterface reader) {
        super.deserialize(reader);
        guestId = reader.readInt();
        if (guestId < 0)
            throw new RuntimeException("Forbidden value on guestId = " + guestId + ", it doesn't respect the following condition : guestId < 0");
    }
    
}

File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/PartyCancelInvitationMessage.scala





















// Generated on 05/03/2013 10:34:46
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PartyCancelInvitationMessage extends AbstractPartyMessage {
  override val messageId = 6254

  protected trait Message extends AbstractPartyMessage.Message with Serializable {
    override val messageId = PartyCancelInvitationMessage.messageId

    val guestId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(guestId)
    }

  }

  protected def create(partyId: Int, guestId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val partyId = buf.readInt()
    val guestId = buf.readInt()

    create(partyId, guestId)
  }

}

object PartyCancelInvitationMessage extends PartyCancelInvitationMessage {
  case class MessageImpl(partyId: Int, guestId: Int) extends Message
  type Target = MessageImpl

  override def create(partyId: Int, guestId: Int) = MessageImpl(partyId, guestId)

  def apply(partyId: Int, guestId: Int) = create(partyId, guestId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

